src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -590,26 +683,100 @@ public override string ToString()
 
         public override uint SizeOf => _size;
 
-        public override void Dispatch(Cci.MetadataVisitor visitor)
-        {
-            visitor.Visit(this);
-        }
-
-        public string Name => _alignment == 1 ?
+        public override string Name => _alignment == 1 ?
             $"__StaticArrayInitTypeSize={_size}" :
             $"__StaticArrayInitTypeSize={_size}_Align={_alignment}";
 
-        public Cci.ITypeDefinition ContainingTypeDefinition => _containingType;
+        public override Cci.ITypeDefinition ContainingTypeDefinition => _containingType;
 
-        public Cci.TypeMemberVisibility Visibility => Cci.TypeMemberVisibility.Assembly;
+        public override Cci.TypeMemberVisibility Visibility => Cci.TypeMemberVisibility.Assembly;
 
         public override bool IsValueType => true;
+    }
 
-        public Cci.ITypeReference GetContainingType(EmitContext context) => _containingType;
+    /// <summary>
+    /// A type synthesized for each eligible string literal to hold the lazily-initialized string.
+    ///
+    /// https://github.com/dotnet/roslyn/blob/main/docs/features/string-literals-data-section.md
+    /// </summary>
+    internal sealed class DataSectionStringType : NestedTypeDefinition
+    {
+        private readonly string _name;
+        private readonly PrivateImplementationDetails _containingType;
+        private readonly ImmutableArray<Cci.IFieldDefinition> _fields;
+        private readonly ImmutableArray<Cci.IMethodDefinition> _methods;
+
+        public DataSectionStringType(
+            ITokenDeferral module,
+            string name,
+            PrivateImplementationDetails containingType,
+            MappedField dataField,
+            Cci.IMethodDefinition bytesToStringHelper,
+            Cci.ITypeReference systemStringType,
```

Comment:
>Cci.ITypeReference systemStringType [](http://example.com/codeflow?start=12&length=35)

I think we should be able to remove the need to thread string type through many layers of method calls. For example, string is a platform type, it can be obtained from a module builder, a module builder is available to ```PrivateImplementationDetails```, which is passed here.

Reply:
Also, it looks like we are passing module builder here as ```ITokenDeferral```. This is another redundancy which I would prefer to get rid of